### DISCUSSION TITLE
So what is happening here / Languages

### DISCUSSION THREAD

**Discussion by oo777** [ggml-org/whisper.cpp#1593] [Q&A]
Very interesting project here! But what is going on? OpenAi developed a speech recognition model named Whisper, currently Whisper-3. And ggerganov turns it into it's own machine learning model named ggml written in C++? And whisper.cpp enables the usage this ggml model with a C-style API and specific platform development like Cuda on Linux? If so, where is the link/integration with the whisper model from OpenAi (https://github.com/openai/whisper) within the whisper.cpp project? 
Finally: Does this ggml model support all the languages like the Whisper-3 model (English, Spanish, ...)

**Comment by bobqianic** [comment]
Emmm...

<img width="616" alt="image" src="https://github.com/ggerganov/whisper.cpp/assets/129547291/70569041-6d4c-4613-a686-046a2b853418">

> Does this ggml model support all the languages like the Whisper-3 model (English, Spanish, ...)

It is consistent with OpenAI. 

> And ggerganov turns it into it's own machine learning model named ggml written in C++? 

> And whisper.cpp enables the usage this ggml model with a C-style API and specific platform development like Cuda on Linux?

Also, model weights are model weights, formats are formats, and the inference framework is the inference framework. **3** completely separate things...
